frame/system/src/lib.rs

```diff
@@ -464,9 +464,15 @@ decl_module! {
 			storage::unhashed::put_raw(well_known_keys::HEAP_PAGES, &pages.encode());
 		}
 
-		/// Set the new runtime code.
+		/// Determine whether or not it is possible to update the code.
+		///
+		/// This function has no side effects and is idempotent, but is fairly
+		/// heavy. It is automatically called by `set_code`; in most cases,
+		/// a direct call to `set_code` is preferable. It is useful to call
+		/// `can_set_code` when it is desirable to perform the appropriate
+		/// runtime checks without actually changing the code yet.
 		#[weight = SimpleDispatchInfo::FixedOperational(200_000)]
-		pub fn set_code(origin, code: Vec<u8>) {
+		pub fn can_set_code(origin, code: Vec<u8>) {
```

Comment:
You only declare dispatchable functions inside `decl_module!`.

This is not an dispatchable function and you also don't want that the function sets `:code` in the storage.

Just put the function into the impl block of the module.